apps/core/scenes/search/Placepickers.js

```diff
@@ -9,39 +9,49 @@ import {
 import { Icon, StyleSheet } from '@kiwicom/universal-components';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
 
-import { withSearchContext, type SearchContextState } from './SearchContext';
+import {
+  withSearchContext,
+  type SearchContextState,
+  type ModalTypes,
+} from './SearchContext';
+import { MODAL_TYPE } from './SearchConstants';
 
 type Props = {|
   +travelFrom: string,
   +travelTo: string,
   +handlePlaceSwitchPress: () => void,
+  +setModalType: ModalTypes => void,
 |};
 
-const handlePlacePress = () => {
-  console.log('TODO'); // eslint-disable-line no-console
-};
+function createHandler(callback: ModalTypes => void, data: ModalTypes) {
+  return () => {
+    callback(data);
+  };
+}
 
-const Placepickers = (props: Props) => (
-  <View>
-    <TripInput
-      onPress={handlePlacePress}
-      label="From"
-      icon={<Icon name="airplane-takeoff" />}
-      value={props.travelFrom}
-    />
-    <TouchableWithoutFeedback onPress={props.handlePlaceSwitchPress}>
-      <View style={styles.placeSwitch}>
-        <Icon name="replace" color="#7F91A8" />
-      </View>
-    </TouchableWithoutFeedback>
-    <TripInput
-      onPress={handlePlacePress}
-      label="To"
-      icon={<Icon name="airplane-landing" />}
-      value={props.travelTo}
-    />
-  </View>
-);
+const Placepickers = (props: Props) => {
+  return (
+    <View>
+      <TripInput
+        onPress={createHandler(props.setModalType, MODAL_TYPE.ORIGIN)}
+        label="From"
+        icon={<Icon name="airplane-takeoff" />}
+        value={props.travelFrom}
+      />
+      <TouchableWithoutFeedback onPress={props.handlePlaceSwitchPress}>
+        <View style={styles.placeSwitch}>
+          <Icon name="replace" color="#7F91A8" />
+        </View>
+      </TouchableWithoutFeedback>
+      <TripInput
+        onPress={createHandler(props.setModalType, MODAL_TYPE.DESTINATION)}
```

Comment:
Why this approach over just creating a class and using the class method which will already have access to the `setModalType`

Reply:
You mean create a class from PlacePicker? 
And then inside create something like `this.handleFromClick` and `this.handleToClick`.
Do I understand right?

Reply:
Yes

Reply:
```
class PlacePickers extends React.Component<Props> {
	onPress = () => {this.props.setModalType(type)}
	render() {
		return ()
	}
}
```

Reply:
OK it is a good idea so I will change it

Reply:
Done